java/jdbc/src/main/java/io/vitess/jdbc/VitessVTGateManager.java

```diff
@@ -174,45 +178,56 @@ private static void actuallyCloseRefreshedConnection(final VTGateConnection old)
    */
   private static VTGateConnection getVtGateConn(VitessJDBCUrl.HostInfo hostInfo,
-      VitessConnection connection) {
+                                                VitessConnection connection) {
     final Context context = connection.createContext(connection.getTimeout());
     RetryingInterceptorConfig retryingConfig = getRetryingInterceptorConfig(connection);
     if (connection.getUseSSL()) {
-      final String keyStorePath = connection.getKeyStore() != null ? connection.getKeyStore()
-          : System.getProperty(Constants.Property.KEYSTORE_FULL);
-      final String keyStorePassword =
-          connection.getKeyStorePassword() != null ? connection.getKeyStorePassword()
-              : System.getProperty(Constants.Property.KEYSTORE_PASSWORD_FULL);
-      final String keyAlias = connection.getKeyAlias() != null ? connection.getKeyAlias()
-          : System.getProperty(Constants.Property.KEY_ALIAS_FULL);
-      final String keyPassword = connection.getKeyPassword() != null ? connection.getKeyPassword()
-          : System.getProperty(Constants.Property.KEY_PASSWORD_FULL);
-      final String trustStorePath = connection.getTrustStore() != null ? connection.getTrustStore()
-          : System.getProperty(Constants.Property.TRUSTSTORE_FULL);
-      final String trustStorePassword =
-          connection.getTrustStorePassword() != null ? connection.getTrustStorePassword()
-              : System.getProperty(Constants.Property.TRUSTSTORE_PASSWORD_FULL);
-      final String trustAlias = connection.getTrustAlias() != null ? connection.getTrustAlias()
-          : System.getProperty(Constants.Property.TRUST_ALIAS_FULL);
-
-      final TlsOptions tlsOptions = new TlsOptions().keyStorePath(keyStorePath)
-          .keyStorePassword(keyStorePassword).keyAlias(keyAlias).keyPassword(keyPassword)
-          .trustStorePath(trustStorePath).trustStorePassword(trustStorePassword)
-          .trustAlias(trustAlias);
-
-      return new RefreshableVTGateConnection(
-          new GrpcClientFactory(retryingConfig).createTls(context, hostInfo.toString(), tlsOptions),
-          keyStorePath, trustStorePath);
+      TlsOptions tlsOptions = getTlsOptions(connection);
+      RpcClient rpcClient = new GrpcClientFactory(retryingConfig)
+          .createTls(context, hostInfo.toString(), tlsOptions);
+      String path = tlsOptions.getTrustStore().getPath();
+      return new RefreshableVTGateConnection(rpcClient, tlsOptions.getKeyStore().getPath(), path);
     } else {
-      return new VTGateConnection(
-          new GrpcClientFactory(retryingConfig).create(context, hostInfo.toString()));
+      RpcClient client = new GrpcClientFactory(retryingConfig).create(context, hostInfo.toString());
+      return new VTGateConnection(client);
     }
   }
 
+  private static TlsOptions getTlsOptions(VitessConnection con) {
+    String keyStorePath = nullIf(con.getKeyStore(), getProperty(Property.KEYSTORE_FULL));
```

Comment:
:nit: If we are already changing this. A more java 8 way to do this would be:

```
      Optional
          .ofNullable(con.getKeyStore())
          .orElse(getProperty(Property.KEYSTORE_FULL))
```

Reply:
Meh... Call me old-school, but I like that my way of doing it is shorter and no extra objects are created.